main.go

```diff
@@ -161,65 +167,118 @@ func NewNSMClient(ctx context.Context, rootConf *config.Config) networkservice.N
 		spiffejwt.TokenGeneratorFunc(source, rootConf.MaxTokenLifetime),
 		clientCC,
 		sendfd.NewClient(),
+		token.NewClient(),
 	)
 }
 
+// Connection is a return type for RunClient
+type Connection struct {
+	Client     networkservice.NetworkServiceClient
+	Connection *networkservice.Connection
+	Cancel     context.CancelFunc
+}
+
 // RunClient - runs a client application with passed configuration over a client to Network Service Manager
-func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*networkservice.Connection, error) {
+func RunClient(ctx context.Context, rootConf *config.Config, nsmClient networkservice.NetworkServiceClient) ([]*Connection, error) {
 	// Validate config parameters
 	if err := rootConf.IsValid(); err != nil {
-		return []*networkservice.Connection{}, err
+		return nil, err
+	}
+
+	var requestConfigs []*requestConfig
+	for i := range rootConf.NetworkServices {
+		nsConf := &rootConf.NetworkServices[i]
+		if err := nsConf.MergeWithConfigOptions(rootConf); err != nil {
+			log.Entry(ctx).Errorf("error during nsmClient config aggregation: %v", err)
+			return nil, err
+		}
+		requestConfigs = appendNSConf(nsConf, requestConfigs)
 	}
 
 	// ********************************************************************************
 	// Initiate connections
 	// ********************************************************************************
 
 	// A list of cleanup operations
-	connections := []*networkservice.Connection{}
-
-	for idx, clientConf := range rootConf.NetworkServices {
-		err := clientConf.MergeWithConfigOptions(rootConf)
-		if err != nil {
-			log.Entry(ctx).Errorf("error during nsmClient config aggregation %v", err)
-			return connections, err
-		}
-		// We need update
-		outgoingMechanism := &networkservice.Mechanism{
-			Cls:        cls.LOCAL,
-			Type:       clientConf.Mechanism,
-			Parameters: map[string]string{},
-		}
-		switch clientConf.Mechanism {
-		case kernel.MECHANISM:
-			outgoingMechanism.Parameters[common.InterfaceNameKey] = clientConf.Path[0]
-			fileURL := &url.URL{Scheme: "file", Path: "/proc/self/ns/net"}
-			kernel.ToMechanism(outgoingMechanism).SetNetNSURL(fileURL.String())
-
-		case memif.MECHANISM:
-			outgoingMechanism.Parameters[memif.SocketFilename] = path.Join(clientConf.Path...)
+	var connections []*Connection
+	for _, requestConfig := range requestConfigs {
+		var clients []networkservice.NetworkServiceClient
+		for _, nsConf := range requestConfig.nsConfs {
```

Comment:
Why scenario when `len(requestConfig.nsConfs) > 0` is valid? 
Could you provide an example when `len(requestConfig.nsConfs)` is more then 1?
What if requestConfig.nsConfs == 0?

I guess we should cover these cases.

Reply:
It is for the case when client provides multiple mechanism preferences for the single request.
Actually I guess it is not very useful, removed and replaced with single-mechanism-preference requests.